tests/lit-tests/3295.ispc

```diff
@@ -0,0 +1,20 @@
+// This test checks that the compiler reports an error when a pointer type is
+// used with unsupported binary operators.
+
+// RUN: not %{ispc} --target=host --nowrap %s -o - 2>&1 | FileCheck %s
+
+// CHECK-NOT: Error: Assertion failed
+// CHECK-COUNT-3: Error: Unsupported binary operator "%" for pointer type: "uniform int32 * uniform"
+// CHECK-COUNT-3: rror: First operand to binary operator "%" is of invalid type "uniform int32 * uniform".
+
+export uniform int gather_ints_buffer(
+    uniform int values[],
+    uniform int indices[],
+    uniform int buffer[],
+    uniform int count,
+    uniform int front_self_range)
+{
+    assume(&values[0] % programCount == 0);
+    assume(&indices[0] % programCount == 0);
+    assume(&buffer[0] % programCount == 0);
```

Comment:
The test can be simplified. Parameters `count` and `front_self_range` are not used and the test checks 3 times the same operation.

Reply:
Yes, sure. I have fixed it.